examples/optimizers/Main.hs

```diff
@@ -0,0 +1,226 @@
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE BlockArguments #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Main where
```

Comment:
any particular reason why this isn't `Torch.Optimizers`?

Reply:
I read the description of your PR again, and I think I understand why you want to have it separate at this point